fideslog/api/models/analytics_event.py

```diff
@@ -81,12 +93,25 @@ def check_not_an_email_address(cls, value: str) -> str:
         return value
 
     @validator("endpoint")
-    def check_no_hostname(cls, value: str) -> str:
+    def validate_endpoint_format(cls, value: str) -> str:
         """
-        Ensure that endpoints contain only the URL path.
+        Ensure that endpoint contains method and path, and that path component contains only the URL path.
         """
 
-        return urlparse(value).path
+        endpoint_components = value.split(":")
+        assert (
+            len(endpoint_components) == 2
+        ), "endpoint must contain only the HTTP method and URL path, delimited by a colon"
+        endpoint_components[0] = endpoint_components[0].strip().upper()
+        assert (
+            endpoint_components[0].strip() in ALLOWED_HTTP_METHODS
```

Comment:
No need to `.strip()` this again now that you're doing it on L105.
```suggestion
            endpoint_components[0] in ALLOWED_HTTP_METHODS
```